missing inline in header file

Dear all,
looking in folder joint_limit_interface/... I have notice the missing of "inline" keyword for all the functions defined in header file. This can cause linking error, such as "multiple definition of ..." if you have a multiple inclusion of the same header file. Could you fix this bug?
Thank you.
Enrico

I try to better explain: I'm using the function getJointLimits(...) defined in joint_limit_interface/joint_limits_urdf.h the linker error is: 

multiple definition of `joint_limits_interface::getJointLimits(boost::shared_ptr<urdf::Joint const>, joint_limits_interface::JointLimits&)'

the inline keyword (as expected) allows to prevent the error of multiple inclusion generated from the linker. The complier is g++ with the C++11 std.

Enrico

I'm using gcc v4.6, if you want to reproduce the error you can create: 

your_header.h 

#ifndef **YOUR_H**
#define **YOUR_H** 

#include <joint_limits_interface/joint_limits_urdf.h>

#endif

and then include your_header.h in two different header file. 
Your linker will find the multiple definition of all the functions defined in the joint_limits_urdf.h...
Let me know if you are able to reproduce the error.

Enrico

--- a/joint_limits_interface/include/joint_limits_interface/joint_limits_rosparam.h
+++ b/joint_limits_interface/include/joint_limits_interface/joint_limits_rosparam.h
@@ -72,7 +72,7 @@ namespace joint_limits_interface
  * existing values. Values in \p limits not specified in the parameter server remain unchanged.
  * \return True if a limits specification is found (ie. the \p joint_limits/joint_name parameter exists in \p nh), false otherwise.
  */
-bool getJointLimits(const std::string& joint_name, const ros::NodeHandle& nh, JointLimits& limits)
+inline bool getJointLimits(const std::string& joint_name, const ros::NodeHandle& nh, JointLimits& limits)
 {
   // Node handle scoped where the joint limits are defined
   ros::NodeHandle limits_nh;

--- a/joint_limits_interface/include/joint_limits_interface/joint_limits_urdf.h
+++ b/joint_limits_interface/include/joint_limits_interface/joint_limits_urdf.h
@@ -48,7 +48,7 @@ namespace joint_limits_interface
  * values. Values in \e limits not present in \e urdf_joint remain unchanged.
  * \return True if \e urdf_joint has a valid limits specification, false otherwise.
  */
-bool getJointLimits(boost::shared_ptr<const urdf::Joint> urdf_joint, JointLimits& limits)
+inline bool getJointLimits(boost::shared_ptr<const urdf::Joint> urdf_joint, JointLimits& limits)
 {
   if (!urdf_joint || !urdf_joint->limits)
   {
@@ -84,7 +84,7 @@ bool getJointLimits(boost::shared_ptr<const urdf::Joint> urdf_joint, JointLimits
  * \param[out] soft_limits Where URDF soft joint limit data gets written into.
  * \return True if \e urdf_joint has a valid soft limits specification, false otherwise.
  */
-bool getSoftJointLimits(boost::shared_ptr<const urdf::Joint> urdf_joint, SoftJointLimits& soft_limits)
+inline bool getSoftJointLimits(boost::shared_ptr<const urdf::Joint> urdf_joint, SoftJointLimits& soft_limits)
 {
   if (!urdf_joint || !urdf_joint->safety)
   {

